frontend/packages/dev-console/src/components/pipelines/pipeline-overview/PipelineOverview.tsx

```diff
@@ -0,0 +1,100 @@
+import * as _ from 'lodash';
+import { ListGroup } from 'patternfly-react';
+import { Grid, GridItem } from '@patternfly/react-core';
+import {
+  SidebarSectionHeading,
+  ResourceLink,
+  resourcePath,
+} from '@console/internal/components/utils';
+import { PipelineRunModel, PipelineModel } from '@console/dev-console/src/models';
+import { Link } from 'react-router-dom';
+import { fromNow } from '@console/internal/components/utils/datetime';
+import { pipelineRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
+import TriggerLastRunButton from './TriggerLastRunButton';
+
+const noOfPipelineRunsToshow = 3;
+
+type PipelineRunItemProps = {
+  pipelineRun: K8sResourceKind;
+};
+
+export const PipelineRunItem: React.FC<PipelineRunItemProps> = ({ pipelineRun }) => {
+  const {
+    metadata: { name, namespace, creationTimestamp },
+    status: { completionTimestamp, startTimestamp },
+  } = pipelineRun;
+  const path = resourcePath(referenceForModel(PipelineRunModel), name, namespace);
+  const lastUpdated = completionTimestamp || startTimestamp || creationTimestamp;
+  return (
+    <li className="list-group-item">
+      <Grid gutter="md">
+        <GridItem span={6}>
+          <Link to={`${path}`}>{name}</Link>
+          {lastUpdated && <span className="text-muted">&nbsp;({fromNow(lastUpdated)})</span>}
+        </GridItem>
+        <GridItem span={3}>
+          <Status status={pipelineRunFilterReducer(pipelineRun)} />
+        </GridItem>
+        <GridItem span={3} className="text-right">
+          <Link to={`${path}/logs`}>View Logs</Link>
+        </GridItem>
+      </Grid>
+    </li>
+  );
+};
+
+type PipelinesOverviewProps = {
+  item: OverviewItem & {
+    pipeline: K8sResourceKind[];
+    pipelineRuns: K8sResourceKind[];
+  };
+};
+
+const PipelinesOverview: React.FC<PipelinesOverviewProps> = ({
+  item: {
+    pipeline: [pipeline],
+    pipelineRuns,
+  },
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+  return (
+    <React.Fragment>
+      <SidebarSectionHeading text="Pipeline Run" style={{ display: 'inline-block' }} />
+      {_.size(pipelineRuns) > noOfPipelineRunsToshow && (
+        <Link
+          className="sidebar__section-view-all"
+          to={`${resourcePath(referenceForModel(PipelineModel), name, namespace)}/Runs`}
+        >
+          {`View All (${_.size(pipelineRuns)})`}
+        </Link>
+      )}
+      <ListGroup componentClass="ul">
+        <li className="list-group-item">
+          <Grid gutter="md">
+            <GridItem span={8}>
+              <ResourceLink
+                inline
+                kind={referenceForModel(PipelineModel)}
+                name={name}
+                namespace={namespace}
+              />
+            </GridItem>
+            <GridItem span={4} className="text-right">
+              <TriggerLastRunButton pipeline={pipeline} />
+            </GridItem>
+          </Grid>
```

Comment:
This Grid is unbalanced - There is a notable gap of whitespace between the end of your first GridItem and the TriggerButton. This space could be used for the title when there are longer pipeline names.

Would recommend using flex to close the gap. A minor set of padding should be all you need to buffer the title from the button, otherwise the title should be allowed to expand and use the space as it needs.

Look at the way the Builds section is doing the title and you'll see the gap is more effectively used.

Reply:
fixed in https://github.com/openshift/console/pull/3056/commits/8e27cbfc8c376c202bd2471c788bf0401b8dcb9d